lib/fog/vsphere/requests/compute/list_resource_pools.rb

```diff
@@ -15,20 +15,45 @@ def list_resource_pools(filters = { })
 
         # root ResourcePool + Children if they exists
         def list_raw_resource_pools(cluster)
-          [cluster.resourcePool, cluster.resourcePool.resourcePool].flatten
+          pools = []
+          traverse_raw_resource_pools(pools,cluster.resourcePool)
+          pools.uniq
         end
 
-        def resource_pool_attributes resource_pool, cluster, datacenter
+        def traverse_raw_resource_pools(pools, rp)
+          if rp != nil
```

Comment:
It's more common to just do `if rp` rather than checking if it's not equal `nil`.